mixin/smarty-v2@1/mixin.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * Auto-register "templates/" folder.
+ *
+ * @mixinName smarty-v2
+ * @mixinVersion 1.0.0
+ * @since 5.58
+ *
+ * @param CRM_Extension_MixInfo $mixInfo
+ *   On newer deployments, this will be an instance of MixInfo. On older deployments, Civix may polyfill with a work-a-like.
+ * @param \CRM_Extension_BootCache $bootCache
+ *   On newer deployments, this will be an instance of MixInfo. On older deployments, Civix may polyfill with a work-a-like.
+ */
+return function ($mixInfo, $bootCache) {
```

Comment:
so the lack of type hints here - do we need to avoid that because Civix supports php 5.x (& if so should it?)

Reply:
It's more about the Civi version than the PHP version.

When deploying the mixin on 5.45+, the objects are literally instances of `CRM_Extension_MixInfo` and `CRM_Extension_BootCache`, so the hard typehint would work there. But when deploying the mixin on an older version of CiviCRM, it uses a polyfill. Each extension bundles its own copy of `polyfill.php`. In order to allow multiple copies of `polyfill.php` to coexist, they use anonymous classes. [This is the anonymous work-a-like](https://github.com/civicrm/civicrm-core/blob/f55a0e7978c92e76c18cd613ece11489052b5b33/mixin/polyfill.php#L36). Hence, the soft typehint.

Reply:
that makes sense